### ISSUE TITLE
Suggestion: Motif Story Engine

### ISSUE THREAD

**Issue by GuardianTempest** [overbo/randm#3]
This one is a paid product unlike freebies such as Tricube Tales or [One Page Solo Engine](https://inflatablestudios.itch.io/one-page-solo-engine).
https://www.drivethrurpg.com/product/316073/Motif-Story-Engine-Solo-and-GMLite-Roleplaying-Toolkit

I recommend just focusing on the core mechanic (3d6 representing Answer, Degree and generic Flavor), the other patches and tweaks would bog things down. The core mechanic is already unique since it also provides variations on how the Yes/No answer comes out. Is it a straight forward answer? Is it a yes that's still hindering? How does that No factor in the surroundings?

**Comment by overbo** [comment]
I love the suggestions, but for every engine that I show in randm-solo, I got permission from the author.  I won't add any tools to randm solo without prior permission from whoever made it.  If someone knows where I can best find the author, I've always been glad to ask, and no one ever said no when the solo RPG community was centered in Google+.

I think the right way to do this would be to allow import of engine via some sort json structure import just like we do with tables.  If I do that, and some clever person can distill the rules of a solo engine like 1pse, motif, tricube tales, etc., they can adapt the rules for personal use.

Right now, the data back-end for a table structure isn't all that great.  It's really designed for a 1:1 representation of a random result.  So if you have six possible entries in a table, and five of them are the same, the data structure looks like this:

``[same], [same], [same], [same], [same], [different]``

but that approach is hindering when there's a different probability curve in front of the table.  1d12 has a straightforward probability curve, whereas 2d6 has a much different one -- more results clustered near 7 would be expected with 2d6 with probability decaying the further one gets from 7, whereas 1d12 should show a pretty equal distribution of probability.

So tl;dr if I can code a data table structure that permits for random results from 2d6 die distribution style instead of a 1d12 die distribution style, adding multiple engines becomes easy.  Ideally, it's just another table structure.

**Comment by overbo** [comment]
1pse and motif are both cc-by-nc-sa and randm-solo is cc-by-nc.  per license rules of CC, it looks like I can combine cc-by-nc and cc-by-nc-sa, so it looks like permission is already granted for similar CC licenses. ( see faq link below ).

1pse is 3 pages, not a ton of logic, so it should be fairly straightforward to include.  i'm guessing that it would be added towards the later part of the v096 dev cycle.  because i like to be polite and being polite is free, i'll reach out to the author to say that i intend to do this, and give them a chance to say 'no thank you' first, because that's how i roll.

it looks like motif is $12, so i'd have to spend money to even think about including it for release.  i'll admit that i have a lot of solo products, but it doesn't look like i own motif.  but the license is right, i'll reach out to the motif creators and see if they are ok with it.  if they say yes, i'll throw down the $12 and evaluate it, and that would dictate where it goes in the dev cycle.  knowing nothing about this one, v097 would be the absolute earliest possibility if everything went well.

tricube tales isn't licensed in CC, so I'm assuming full copyright terms apply.  i own tricube tales as a customer and it looks fairly straightforward to implement, but it also looks like the sauce is the random tables themselves.  releasing those tables into a free product eats their money, so i wouldn't hold my breath on them saying "yes" to the idea of incorporating the meat of their product into randm solo.

https://creativecommons.org/faq/#can-i-combine-material-under-different-creative-commons-licenses-in-my-work
